src/include/migraphx/instruction_ref.hpp

```diff
@@ -32,7 +32,44 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 struct instruction;
+#if defined(_WIN32) && !defined(NDEBUG)
+struct instruction_ref : std::list<instruction>::iterator
+{
+    using instruction_iter       = std::list<instruction>::iterator;
+    using instruction_const_iter = std::list<instruction>::const_iterator;
+
+    instruction_ref() = default;
+    instruction_ref(const instruction_iter& other)
+    {
+        _Ptr = other._Ptr;
+        _Adopt(other._Getcont());
```

Comment:
Cant we use the base copy constructor for this?

Reply:
Good idea, done.